sonar-db/src/main/java/org/sonar/db/permission/PermissionTemplateDao.java

```diff
@@ -179,6 +186,23 @@ public PermissionTemplateDto selectByUuidWithUserAndGroupPermissions(String temp
     }
   }
 
+  public List<PermissionTemplateDto> selectPermissionTemplates(DbSession session, @Nullable String nameQuery, int offset, int limit) {
+    if (nameQuery != null) {
+      String wildcard = "%";
+      nameQuery = wildcard + nameQuery.toUpperCase() + wildcard;
+    }
+    return mapper(session).selectPermissionTemplates(nameQuery, new RowBounds(offset, limit));
+  }
+
+  public int countPermissionTemplates(DbSession dbSession, @Nullable String nameQuery) {
+    if (nameQuery != null) {
+      String wildcard = "%";
+      nameQuery = wildcard + nameQuery.toUpperCase() + wildcard;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "nameQuery". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)